emcc.py

```diff
@@ -3872,7 +3872,7 @@ def modularize():
     diagnostics.warning('emcc', 'EXPORT_NAME should not be named "config" when targeting Safari')
 
   src = '''
-%(maybe_async)sfunction(%(EXPORT_NAME)s = {})  {
+%(maybe_async)s (Module = {}) => {
```

Comment:
The logic is pretty complex in this function and I can't quite follow it. Can you explain why this change is valid?

Reply:
In the case of MODULARIZE we no longer `var Module = ..` in `shell.js` because we know we can are going to receive it via the (possibly default) argument here.

In `shell.js` now define `var {{ EXPORT_NAME }} ..` locally just in case the user wants to refer to it locally by this name.. I've added a comment and a TODO about that.